FirebaseMessaging/Sources/FIRMessaging.m

```diff
@@ -437,9 +437,7 @@ - (void)handleIncomingLinkIfNeededFromMessage:(NSDictionary *)message {
     // Similarly, |application:openURL:sourceApplication:annotation:| will also always be called,
     // due to the default swizzling done by FIRAAppDelegateProxy in Firebase Analytics
   } else if ([appDelegate respondsToSelector:openURLWithSourceApplicationSelector]) {
-// TODO(Xcode 15): When Xcode 15 is the minimum supported Xcode version, it will be unnecessary to
-// check if `TARGET_OS_VISION` is defined.
-#if TARGET_OS_IOS && (!defined(TARGET_OS_VISION) || !TARGET_OS_VISION)
+#if TARGET_OS_IOS
```

Comment:
I didn't try compiling it for visionOS but should this be `#if TARGET_OS_IOS && || !TARGET_OS_VISION` now ([`application:openURL:sourceApplication:annotation:`](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623073-application?language=objc) isn't available on visionOS)?

Reply:
My understanding is that in Xcode 15.2+, TARGET_OS_VISION is never true when TARGET_OS_IOS is.

Reply:
You're right:
> Fixed: When building with the visionOS SDK, TargetConditionals.h incorrectly sets TARGET_OS_IOS to 1 instead of 0. (112716373)

-- [Xcode 15.2 Release Notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-15_2-release-notes#Resolved-Issues)

Sorry, there was some churn around these constants and I wasn't sure.